lib/apis/moray.js

```diff
@@ -631,35 +664,152 @@ Moray.prototype._addFilterOptions = function (params) {
         }
 
         if (splitted.length == 2) {
-            order = splitted[1] || defaultOrder;
+            order = splitted[1] || common.DEFAULT_SORT_ORDER;
         }
 
-        options.sort = {
+        sortOptions.push({
             attribute: property,
             order: order
-        };
-    } else {
-        options.sort = {
-            attribute: '_id',
-            order: 'ASC'
-        };
+        });
+    }
+
+    // If a marker is used, we need to make sure about a few things:
+    //
+    // 1) The resulting data set is sorted on all keys used in the marker
+    //
+    // 2) The sorting used also outputs a data set that is strictly and
+    // totally sorted
+    //
+    // That allows any entry of the resulting data set to be compared to the
+    // entry identified by the marker, and thus the latest item of any result
+    // set can be used as a marker to get the next page of results.
+    if (markerParam) {
+        // If the marker contains keys that are not included in the sort
+        // options, add them to the sort options so that the filter used
+        // to get the data operate on a totally ordered data set.
```

Comment:
Is that the right thing to do?  Should we not just fail the request if the caller provides properties in the marker that are not also required by (present in) the sort order?